decoders/connector/digi-x-on/digi-x-on-sparkfun-iot-node/v1.0.0/payload.test.ts

```diff
@@ -0,0 +1,51 @@
+import { readFileSync } from "fs";
+import { join } from "path";
+import * as ts from "typescript";
+import { beforeEach, describe, expect, test } from "vitest";
+
+import { DataToSend } from "@tago-io/sdk/lib/types";
+
+// Load the JS decoder code
+const decoderFile = readFileSync(join(__dirname, "./payload.js"), "utf8");
+const transpiled = ts.transpile(decoderFile);
+
+let payload: DataToSend[] = [];
+
+(globalThis as any).payload = payload;
+
+describe("SparkFun Decoder (toTagoFormat)", () => {
+  beforeEach(() => {
+    payload = [{ variable: "payload", value: "05127A" }];
+    (globalThis as any).payload = payload;
+  });
+
+  test("Decodes AQI sensor (0x05) with value", () => {
+    eval(transpiled);
+
+    const aqiVar = payload.find(item => item.variable === "AQI");
+    expect(aqiVar).toBeDefined();
+    expect(typeof aqiVar?.value).toBe("number");
+  });
+
+  test("Handles unknown sensor gracefully", () => {
+    payload = [{ variable: "payload", value: "FFFF1234" }];
+    (globalThis as any).payload = payload;
+
+    eval(transpiled);
+
+    const unknown = payload.find(item => item.variable === "unknown_sensor");
+    expect(unknown).toBeDefined();
+    expect(unknown?.value).toBe("FF");
+  });
+
+  test("Handles empty or missing payload", () => {
```

Comment:
The decoder must be able to allow regular non payload variables to pass, otherwise other functionalities will not work properly. This is necessary otherwise if a user attempts to manually add data to the device without decoding, they will not be able to since the "payload" variable won't be present.

Change the remove the functionality the "Handles empty or missing payload" tests for, and add following test type: 

describe("Allow normal variables to pass", () => {
  let payload = [
    { variable: "shallnotbedecoded", value: "04096113950292" },
    { variable: "fport", value: 9 },
  ];
  
  payload = decoderRun(file_path, { payload });
  test("Output Result", () => {
    expect(Array.isArray(payload)).toBe(true);
  });